h/views/api_auth.py

```diff
@@ -31,6 +32,43 @@ def __init__(self, context, request):
     @view_config(request_method='GET',
                  renderer='h:templates/oauth/authorize.html.jinja2')
     def get(self):
+        return self._authorize()
+
+    @view_config(request_method='GET',
+                 request_param='response_mode=web_message',
+                 renderer='h:templates/oauth/authorize.html.jinja2')
+    def get_web_message(self):
+        response = self._authorize()
+
+        if isinstance(response, HTTPFound):
```

Comment:
In both this method and `post_web_message`, it looks like we’re calling something that produces a redirect URL, wraps it up in an HTTP response and returns it, whereupon we *unwrap* it, extract the location, and just use that. I wonder, is there a different abstraction we could use here which wouldn’t require wrapping the location up in a request only to unwrapp it again?